### PR TITLE
Ordered set lattices.

### PR DESCRIPTION
**Pull request by xyclin**
- Anna-side C++ representation.
- Anna-side serialization and deserialization.
- Client-side python representation.
- Anna C++ client interface.
- Python IPC client serialization and deserialization.

### CODE REVIEW

File: include/lattices/lww_pair_lattice.hpp [hunk 1]
```diff
@@ -52,7 +52,7 @@ class LWWPairLattice : public Lattice<TimestampValuePair<T>> {
   }
 
  public:
-  LWWPairLattice() : Lattice<TimestampValuePair<T>>() {}
+  LWWPairLattice() : Lattice<TimestampValuePair<T>>(TimestampValuePair<T>()) {}
```
Comment (cw75) [inline include/lattices/lww_pair_lattice.hpp]: why modify this?

File: include/lattices/vector_clock_pair_lattice.hpp [hunk 2]
```diff
@@ -57,7 +57,8 @@ class CausalPairLattice : public Lattice<VectorClockValuePair<T>> {
   }
 
  public:
-  CausalPairLattice() : Lattice<VectorClockValuePair<T>>() {}
+  CausalPairLattice() :
```
Comment (cw75) [inline include/lattices/vector_clock_pair_lattice.hpp]: why modify this?

File: include/lattices/lww_pair_lattice.hpp (hunk 1 above)
Comment (xyclin) [reply include/lattices/lww_pair_lattice.hpp]: I had to take out the default (zeroarg) constructor for Lattice, which was making bottom lattices in a nongeneralizable way. Here, I'm replacing the call to the default constructor by providing an explicit bottom value. Same for your other comment as well.

File: include/lattices/vector_clock_pair_lattice.hpp (hunk 2 above)
Comment (xyclin) [reply include/lattices/vector_clock_pair_lattice.hpp]: See other comment.

File: include/lattices/lww_pair_lattice.hpp (hunk 1 above)
Comment (cw75) [reply include/lattices/lww_pair_lattice.hpp]: got it

**Comment by xyclin** [comment]
@cw75  can you approve again? i had to force-push after rebasing

**Pull request merged**